plugin.video.iplayerwww/resources/lib/ipwww_video.py

```diff
@@ -813,50 +785,23 @@ def AddAvailableLiveStreamsDirectory(name, channelname, iconimage):
         iconimage: only used for displaying the channel.
         channelname: determines which channel is queried.
     """
-    retlist = []
+    streams = ParseLiveStreams(channelname, '')
 
-    #for device in ['iptv-all', 'apple-ipad-hls']:
-    for device in ['iptv-all']:
-        NEW_URL = "http://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/%s/vpid/%s/transferformat/hls?cb=%d" % (device, channelname, random.randrange(10000,99999)) 
-        html = OpenURL(NEW_URL)
-        match = re.compile(
-            'media.+?bitrate="(.+?)".+?encoding="(.+?)".+?connection.+?href="(.+?)".+?supplier="(.+?)".+?transferFormat="(.+?)"'
-            ).findall(html)
-        urls = []
-        for bitrate, encoding, url, supplier, transfer_format in match:
-            urls.append(url)
-        urls = list(set(urls)) #unique
-        for NEW_URL in urls:
-            html = OpenURL(NEW_URL)
-            match = re.compile('#EXT-X-STREAM-INF:PROGRAM-ID=(.+?),BANDWIDTH=(.+?),CODECS="(.*?)",RESOLUTION=(.+?)\s*(.+?.m3u8)').findall(html)
-            for id, bandwidth, codecs, resolution, url in match:
-                bitrate = int(int(bandwidth)/1000.0)
-                retlist.append((supplier, bitrate, url, device))
-
-    #for device in ['abr_hdtv', 'hls_tablet']:
-    for device in ['abr_hdtv']:
-        for supplier in ['ak', 'llnw']:
-            NEW_URL = "http://a.files.bbci.co.uk/media/live/manifesto/audio_video/simulcast/hls/uk/%s/%s/%s.m3u8" % (device, supplier, channelname)
-            html = OpenURL(NEW_URL)
-            match = re.compile('#EXT-X-STREAM-INF:PROGRAM-ID=(.+?),BANDWIDTH=(.+?),CODECS="(.*?)",RESOLUTION=(.+?)\s*(.+?.m3u8)').findall(html)
-            for id, bandwidth, codecs, resolution, url in match:
-                bitrate = int(int(bandwidth)/1000.0)
-                retlist.append((supplier, bitrate, url, device))
-
-    retlist = sorted(retlist, key=itemgetter(1,0), reverse=True)
-
-    for (supplier, bitrate, url, device) in retlist:
-
-        if bitrate > 2100:
+    # Add each stream to the Kodi selection menu.
+    for id, bitrate, codecs, resolution, url, provider_name in streams:
+        # For easier selection use colors to indicate high and low bitrate streams
+        if bitrate > 2.1:
             color = 'green'
-        elif bitrate > 1000:
+        elif bitrate > 1.0:
             color = 'yellow'
-        elif bitrate > 600:
+        elif bitrate > 0.6:
             color = 'orange'
         else:
             color = 'red'
-        # title = name + ' - [I][COLOR %s]%0.1f Mbps[/COLOR] [COLOR white]%s[/COLOR] [COLOR grey]%s[/COLOR][/I]' % (color, bitrate/1000.0, supplier, device)
```

Comment:
I would recommend to not use color names since those are skin dependent... better take the real AARRGGBB values.